scripts/ci-generate-publish-metadata.sh

```diff
@@ -21,6 +21,8 @@ GIT_SHA=$(git rev-parse --short HEAD)
 CURRENT_DATE=$(date +%Y-%m-%d)
 echo "vector_build_desc=${GIT_SHA} ${CURRENT_DATE}" >> "${GITHUB_OUTPUT}"
 
+echo "vector_git_sha=${GIT_SHA}" >> "${GITHUB_OUTPUT}"
```

Comment:
Why is this needed now?

Reply:
Same answer as https://github.com/vectordotdev/vector/pull/16281#discussion_r1137833090

Reply:
We could compute it later on closer to where it is needed, but this seemed pretty harmless and saves the repeated effort.